exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/AbstractDataCollector.java

```diff
@@ -116,4 +120,16 @@ public void close() throws Exception {
     AutoCloseables.close(buffers);
   }
 
+  /** {@inheritDoc} */
+  @Override
+  public BufferAllocator getAllocator() {
+    return this.ownerAllocator;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void setAllocator(BufferAllocator allocator) {
+    this.ownerAllocator = allocator;
```

Comment:
Can passed allocator be `null`? If yes, `ownerAllocator` should not probably be overwritten. If no, please add an `assert` or better `Preconditions` check.

Reply:
I wondered about the same; the only reasons for the caller to pass a null, is either a bug or allocator unregistration. I guess for now, I'll add an assert since we do not really need an unregistration step (this will also help us catch erroneous setting of the allocator).

Reply:
I'd recommend `Preconditions` over `assert`.

Reply:
sure.